elm-canvas/elm-canvas.js

```diff
@@ -12,23 +12,38 @@ customElements.define(
       this.render();
     }
 
+    static get observedAttributes() {
+      return ['canvas.width', 'canvas.height'];
+    }
+
+
+    setCanvasDimensions(w, h){
+        var devicePixelRatio = window.devicePixelRatio || 1;
+        this.canvas.style.width = w;
+        this.canvas.style.height = h;
+        this.canvas.width = w * devicePixelRatio;
+        this.canvas.height = h * devicePixelRatio;
+        this.context.scale(devicePixelRatio, devicePixelRatio);
+    }
+
     connectedCallback() {
       requestAnimationFrame(() => {
         this.canvas = this.querySelector("canvas");
         this.context = this.canvas.getContext("2d");
         this.mounted = true;
 
-        var devicePixelRatio = window.devicePixelRatio || 1;
-        this.canvas.style.width = this.canvas.width;
-        this.canvas.style.height = this.canvas.height;
-        this.canvas.width = this.canvas.width * devicePixelRatio;
-        this.canvas.height = this.canvas.height * devicePixelRatio;
-        this.context.scale(devicePixelRatio, devicePixelRatio);
+        setCanvasDimensions(this.canvas.width, this.canvas.height);
 
         this.render();
       });
     }
 
+    attributeChangedCallback(name, oldValue, newValue) {
+      //we're assuming that only canvas.width or canvas.height can change
+      setCanvasDimensions(this.canvas.width,this.canvas.height)
```

Comment:
```suggestion
      if ((name === 'width' || name === 'height') && oldValue !== newValue) {
        this.setCanvasDimensions(this.width, this.height)
      }
```

We may have to parseInt these, like I mentioned in the comment above. And if this.width doesn't work, use this.getAttribute('width') instead.

Reply:
`this.` was missing too